### ISSUE TITLE
memory allocation error did not halt script from running

### ISSUE THREAD

**Issue by ajkroeg** [glmmTMB/glmmTMB#500]
Sorry in advance that I do not have a reprex for this; it's only happened to me once, and I didn't think to save the output (I've been unable to reproduce it since).  Figured I'd bring it up anyway.

When running a mixed poisson on a large dataset (~358000 observations, 6 variables, lots of interactions and quadratic terms), I encountered the following error:

`Error in sparseHessianFun(env, skipFixedEffects = skipFixedEffects) :Memory allocation fail in function 'MakeADHessObject2'` 

I was running with verbose=TRUE, so I know that this error occurred during one of the "Optimizing tape" iterations.  The error itself is no big deal.  Since it's just a memory error, I'd just unload some stuff from the environment, close programs, etc. and try to run it again (which did indeed work).

The part that concerns me is that this memory error did not cause the script to halt; instead, it looked like the script just started over, did not encounter an error, and proceeded on to generate a final model (call it M2) object, complete with AIC, BIC, log-lik, etc.  However, the returns of the model itself were completely off.  AIC, BIC, etc were something like 3 times lower than an almost identical model (call it M1), when it should have only been a few points higher or lower (only difference between M1 and M2 is that I removed one near-zero random-term).   In addition, DF for M2 was incorrectly reported as being identical to that of M1, when it should have been one less.  This was confirmed when I exited R, closed out some other programs, and re-ran the model; this time, I got no memory error, and the resulting model output was just as I'd expect; reasonable AIC, etc., and correct Df.

**Comment by bbolker** [comment]
I agree that this is unwanted behaviour.  It happens within TMB ([here](https://github.com/kaskr/adcomp/blob/b8095ab1c696b65d54e0b2027cf70e27dd50cc9a/TMB/inst/include/tmb_core.hpp#L1509)); the trapping/error code is [here](https://github.com/kaskr/adcomp/blob/master/TMB/inst/include/tmb_core.hpp#L8-L24), which makes it look like this should throw a regular R error. I don't see any relevant `try()`/`tryCatch()` calls in glmmTMB so ... ??? (You didn't by any chance have any `try()`/`tryCatch()` statements in your code, did you ?)

**Comment by ajkroeg** [comment]
Nope, nothing like that.  I was using the older syntax that I had to use before the mapArgs branch was merged, e.g.:
```
m1str<-glmmTMB(Response ~ v1+ v2+ v3+
                                (1|ID2) +(0+v3|ID)
                                , family=poisson, data=dat ,doFit=F, verbose = T)
m1str$parameters$theta[1] = log(1e3)
m1str$mapArg = list(theta=factor(c(NA,1)))
v1<-glmmTMB:::fitTMB(m1str)
```
but unless `doFit=F` also implies a `tryCatch()` statement, I don't think that'd make a difference...?

**Comment by bbolker** [comment]
I'm hoping @kaskr can shed some light on this ...
